gdal/ogr/ogrsf_frmts/sosi/ogrsosilayer.cpp

```diff
@@ -129,9 +132,30 @@ OGRFeature *OGRSOSILayer::GetNextFeature() {
     short nName, nNumLines;
     long  nNumCoo;
     unsigned short nInfo;
-     // TODO make a parameter that can be sent from command line
+
+    // Used to limit the number of limit the fields to be appended.
+    // Default is that duplicates will appended if no ogrSosiAppendFieldsMap parameter is sent.
     std::map<std::string,std::string> appendFieldsMap;
-    appendFieldsMap.insert(std::pair<std::string,std::string>("BEITEBRUKERID",","));
+
+	// Get ogrSosiAppendFieldsMap and update appendFieldsMap if present;
+    // The input must on this format
+    // -oo ogrSosiAppendFieldsMap="BEITEBRUKERID:,&FIELD2test: &FIELD3test:;"
+    // With the example abouve the field BEITEBRUKERID append character will be ':', for FIELD2test it will space and for FIELD3test it will be ;
+    const char * ogrSosiAppendFieldsMapInput = CSLFetchNameValue(poParent->GetOpenOptions(),"ogrSosiAppendFieldsMap");
+    if (ogrSosiAppendFieldsMapInput != NULL) {
+        char *ogrSosiAppendFieldsMapCopy = strdup(ogrSosiAppendFieldsMapInput);
```

Comment:
will leak memory. and strtok() is not thread safe
Instead you can use
```
CPLStringList aosTokens(CSLTokenizeString2(ogrSosiAppendFieldsMapInput, "&", 0));
for( int i = 0; i < aosTokens.size(); i++ )
{
   // aosTokens[i]
}
```
Do you really need a different separator per field ? That seems to be over-complication to me. If not, then, if you want this to be configurable, a dedicated open option could be used;

Reply:
Thanks, Tried to fix memory leak, renamed oo variable and fixed no need for adding delimiter character if not needed.

-oo appendFieldsMap="BEITEBRUKERID:,&FIELD2test: &FIELD3test:;"
With the example above the field BEITEBRUKERID append character will be ':', for FIELD2test it will space and for FIELD3test it will be ;

-oo appendFieldsMap="BEITEBRUKERID&FIELD2test&FIELD3test"
With the example above the append character will be ',' for fields in the list